src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,107 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+import { useState, useEffect, useCallback } from 'react';
+import { Todo } from '../../types/Todo';
+
+interface TodoFilterProps {
+  todos: Todo[];
+  handleFilterChange: (filteredTodos: Todo[]) => void;
+}
+
+export enum FilterOptions {
+  ALL = 'all',
+  COMPLETED = 'completed',
+  ACTIVE = 'active',
+}
+
+export const TodoFilter: React.FC<TodoFilterProps> = ({
+  todos,
+  handleFilterChange,
+}) => {
```

Comment:
You're initializing state with props. This is generally considered an anti-pattern in React. The props can change, and your component won't reflect these changes because you've copied the props to the local state. Please, consider removing state that duplicates props. More details can be found [here](https://stackoverflow.com/a/50403930) and [here](https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state).